commands/daemon.go

```diff
@@ -122,8 +123,23 @@ func getRepo(req *cmds.Request) (repo.Repo, error) {
 	return repo.OpenFSRepo(getRepoDir(req))
 }
 
-func runAPIAndWait(ctx context.Context, node *node.Node, config *config.Config, req *cmds.Request) error {
-	api := impl.New(node)
+func runAPIAndWait(ctx context.Context, nd *node.Node, config *config.Config, req *cmds.Request) error {
+	api := impl.New(nd)
+	blockTime, mineDelay := nd.MiningTimes()
+	protocolAPI := bapi.New(
```

Comment:
strong suggestion, though i guess not strictly blocking: i think it makes sense to keep all the construction logic together in one place until we have a chance to refactor it. suggest that this protocol api be instantiated in config.build somewhere around https://github.com/filecoin-project/go-filecoin/blob/124cd65964c2d1a55c0816fd0d5700087d1a8099/node/node.go#L403, set on node, and then we can pull it out of node here like we do the porcelainapi.

down the road node just becomes a constructor and container for the various layers: core apis (repo, wallet, cborstore, host, etc), plumbing, porcelain, protocols, and protocol apis.